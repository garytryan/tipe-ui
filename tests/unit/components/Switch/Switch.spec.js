import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import TipeSwitch from '@/components/Switch'

const props = {
  text: 'Share Location',
  name: 'location'
}
describe('Progress.vue', () => {
  it('renders', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        ...props
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
  it('has correct data-tipe-ui attibute', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSwitch')
  })
  it('should have the correct label', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.find('.label').text()).toBe('Share Location')
  })
  it('should have the correct sublabel', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        ...props,
        sublabel: 'Share location while using the app'
      }
    })
    expect(wrapper.find('.sublabel').text()).toBe(
      'Share location while using the app'
    )
  })
  it('should not render with a sublabel', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.find('.sublabel').exists()).toBe(false)
  })
  it('should have the correct value prop', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        ...props,
        value: true
      }
    })
    expect(wrapper.props().value).toBe(true)
  })
  it('should have the correct props', () => {
    const wrapper = shallowMount(TipeSwitch, {
      propsData: {
        ...props,
        sublabel: 'Share location while using the app',
        value: false,
        disabled: true
      }
    })
    expect(wrapper.props().text).toBe('Share Location')
    expect(wrapper.props().sublabel).toBe('Share location while using the app')
    expect(wrapper.props().value).toBe(false)
    expect(wrapper.props().name).toBe('location')
    expect(wrapper.props().disabled).toBe(true)
  })
})
