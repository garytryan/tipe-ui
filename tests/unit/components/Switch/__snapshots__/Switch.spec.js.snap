// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress.vue renders 1`] = `
<div data-tipe-ui="TipeSwitch" class="wrapper">
  <div class="row">
    <label for="Share Location-switch" class="label">Share Location</label>
    <label class="switch">
      <input id="Share Location-switch" type="checkbox" name="location" checked="checked"> <span class="slider round"></span></label>
  </div>
  <!---->
</div>
`;
